src/lsm/groove.zig

```diff
@@ -526,7 +527,7 @@ pub fn GrooveType(
             // output tables until the compaction is complete. (Until then, the output tables may
             // be in the manifest but not yet on disk).
             const snapshot_max = groove.objects.lookup_snapshot_max;
-            assert(snapshot_max == groove.ids.lookup_snapshot_max);
+            assert(!has_id or snapshot_max == groove.ids.lookup_snapshot_max);
```

Comment:
the `has_id` before acts as a comptime short-circuit to avoid accessing `groove.ids` if it doesn't exist. Would something like `if (has_id) { \n assert()` be better?

Reply:
Ah, forgot about that. You can leave it as-is.